testutil/keeper/unit_test_helpers.go

```diff
@@ -101,4 +102,61 @@ func GetIBCQueryKeeperAndCtx(t *testing.T, params InMemKeeperParams) (
 	ctrl := gomock.NewController(t)
 	mocks := NewMockedKeepers(ctrl)
 	return NewInMemIBCQueryKeeper(params, mocks), params.Ctx, ctrl, mocks
+}
+
+// GetMocksForSubmitCrossChainQuery returns mock expectations needed to call SubmitCrossChainQuery()
+func GetMocksForSubmitCrossChainQuery(ctx sdk.Context, mocks *MockedKeepers, 
+	queryId string, senderAddr string) []*gomock.Call {
+	var( errFromNewCap  error )
+	dummyCap := &capabilitytypes.Capability{}
+	return []*gomock.Call{			
+		mocks.MockScopedKeeper.EXPECT().NewCapability(
+			ctx, types.FormatQueryCapabilityIdentifier(queryId, senderAddr),
+		).Return(dummyCap, errFromNewCap).Times(1),
+	}
+}
+
+// setUpForCapabilityQueryingChain returns mock expectations needed to call PruneCrossChainQueryResult()
+func GetMocksForPruneCrossChainQueryResult(ctx sdk.Context, mocks *MockedKeepers, 
+	queryId string, senderAddr string) []*gomock.Call {
+	dummyCap := &capabilitytypes.Capability{}
+	return []*gomock.Call{			
+		mocks.MockScopedKeeper.EXPECT().GetCapability(
+			ctx, types.FormatQueryCapabilityIdentifier(queryId, senderAddr),
+		).Return(dummyCap, true).Times(1),
+		mocks.MockScopedKeeper.EXPECT().AuthenticateCapability(
+			ctx, dummyCap, types.FormatQueryCapabilityIdentifier(queryId, senderAddr),
+		).Return(true).Times(1),
+	}
+}
+
+// GetQueryId returns query identifier
+func GetQueryId(IBCQueryKeeper ibcquerykeeper.Keeper, ctx sdk.Context) string {
+	nextQuerySeq := IBCQueryKeeper.GetNextQuerySequence(ctx)
+	return types.FormatQueryIdentifier(nextQuerySeq)
+}
+
+// GetMsgSubmitCrossChainQuery returns MsgSubmitCrossChainQuery
+func GetMsgSubmitCrossChainQuery(ctx sdk.Context, path string, senderAddr string) types.MsgSubmitCrossChainQuery {
```

Comment:
How about "NewMsgSubmitCrossChainQueryResultForTest" ?

"Get" means retrieving existing results but this function makes new instance